src/Tabs.js

```diff
@@ -0,0 +1,199 @@
+	 * @param {number} index
+	 */
+	addTabByName(label, disabled, index) {
+		if (core.isString(label) && core.isBoolean(disabled)) {
+			let tab = {
+				label: label,
+				disabled: disabled
+			};
+
+			this.addTab(tab, index);
+		}
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	created() {
+		this.tab = this.tabs.findIndex(item => (!item.disabled || !core.isDef(item.disabled)) && !this.disabled);
+
+		if (core.isNull(this.type) && !core.isDefAndNotNull(this.elementClasses)) {
+			this.type = Tabs.TYPES.TABS;
+		}
+
+		this.lastState_ = {
+			tab: this.tab
+		};
+
+		this.on(Tabs.Events.CHANGE_REQUEST, this.defaultChangeRequestFn_, true);
+	}
+
+	/**
+	 * Default `changeRequest` function, sets new state of tabs.
+	 * @param {EventFacade} event
+	 * @protected
+	 */
+	defaultChangeRequestFn_(event) {
+		this.setState_(event.state);
+	}
+
+	/**
+	 * Fires `changeRequest` event.
+	 * @param {Object} state
+	 * @protected
+	 */
+	dispatchRequest_(state) {
+		this.emit(
+			Tabs.Events.CHANGE_REQUEST,
+			{
+				lastState: this.lastState_,
+				state: state,
+				totalTabs: this.tabs.length
+			}
+		);
+	}
+
+	/**
+	 * `onClick` handler for tab items.
+	 * @param {EventFacade} event
+	 */
+	onClickItem(event) {
+		let item = event.delegateTarget;
+
+		event.preventDefault();
+
+		var index = parseInt(item.getAttribute('data-index'));
+
+		this.dispatchRequest_({
+			tab: index
+		});
+	}
+
+	/**
+	 * Removes the tab at the given index from the tabs array.
+	 * @param  {number} index
+	 * @return {Object} Returns the removed tab.
+	 */
+	removeTab(index) {
+		if (core.isNumber(index) && index > -1 && index < this.tabs.length) {
+			return this.tabs.splice(index, 1);
+		}
+	}
+
+	/**
+	 * Set the new tabs state. The state is a payload object
+	 * containing the tab number, e.g. `{tab:1}`.
+	 * @param {Object} state
+	 * @protected
+	 */
+	setState_(state) {
+		this.tab = state.tab;
+
+		this.lastState_ = state;
+	}
+
+	/**
+	 * Disables the tab at the given index in the tabs array.
+	 * @param  {number} index
+	 * @return {boolean} disabled
+	 */
+	setTabDisabled(index, disabled) {
+		if (core.isNumber(index) && core.isBoolean(disabled)) {
+			this.tabs[index].disabled = disabled;
+
+			this.tabs = this.tabs;
+		}
+	}
+}
+Soy.register(Tabs, templates);
+
+Tabs.Events = {
+	CHANGE_REQUEST: 'changeRequest'
+};
+
+/**
+ * Default types used to style the tabs component.
+ */
+Tabs.TYPES = {
+	PILLS: 'pills',
+	TABS: 'tabs'
+};
+
+/**
+ * State definition.
+ * @type {!Object}
+ * @static
+ */
+Tabs.STATE = {
+	/**
+	 * Determines if the whole component should be disabled or not.
+	 * @type {boolean}
+	 * @default false
+	 */
+	disabled: {
+		validator: core.isBoolean,
+		value: false
+	},
+
+	/**
+	 * Tab to display on initial paint.
+	 * @type {number}
+	 * @default Default will be set in the created lifecycle of the compoenent.
+	 */
+	tab: {
+		validator: core.isNumber,
+		value: 0
+	},
+
+	/**
+	 * Tabs array, holding the actual tabs.
```

Comment:
Would be good to mention what can be inside each tab object (saying that label is required, and disabled isn't for example).